midend/coverage.h

```diff
@@ -29,6 +29,10 @@ struct CoverageOptions {
     bool coverTableEntries = false;
     /// Skip tests which do not increase coverage.
     bool onlyCoveringTests = false;
+
+    /// @returns true if and only if at least one of the coverage-collection options (cover*) is
+    /// enabled
+    bool coverageEnabled() const { return coverStatements || coverTableEntries; }
```

Comment:
I've opted to add this getter to make it more likely this will be properly updated when new coverage types are added.

Reply:
Instead, I would add an option to `CoverageOptions` that sets this value true, whenever one of the values is set.

Reply:
I am not sure I understand what you mean. Do you suggest to have a derived boolean argument in the struct? That would seem like needless duplication of information at risk of inconsistency to me.

Reply:
Yes, set the value here: https://github.com/p4lang/p4c/blob/main/backends/p4tools/modules/testgen/options.cpp#L256

You can easily precompute the boolean value here. It will also automatically pick up any new coverage option that might be added. This way you do not need to hard-code all the possible coverage options in the helper.

Reply:
I don't like that all that much. The `CoverageOptions` struct is not strictly bound to be filled by options only, it is not even defined in the headers that process options. Therefore, having a derived attribute adds an invariant to the class that needs to hold when the class values are modified/set. Having the getter needs an invariant too (that it ORs all the `cover*` values), but this one is at least localized to the class itself.

Reply:
You could also add this flag to the testgen options class instead of the options structure.

Reply:
OK, let's do it that way. Although I still think having the member function did make sense.